core/templates/dev/head/pages/dashboard/Dashboard.js

```diff
@@ -92,6 +92,26 @@ oppia.controller('Dashboard', [
     $scope.unresolvedAnswersIconUrl = UrlInterpolationService.getStaticImageUrl(
       '/icons/unresolved_answers.svg');
 
+    // Refreshes the dashboard page whenever the back button is pressed
+    $scope.$watch(function() {
+      return $location.path();
+    }, function(newPath, oldPath) {
+      if (performance.navigation.type !== 1) {
+        if (newPath === '' && oldPath === '') {
+          $location.path('').replace;
```

Comment:
It was working because it was getting called by reference. Mistake from my side sorry. Changed it to a function call.